### PR TITLE
Version 1.0.0

### PR DESCRIPTION
**Pull request by alallier**
Version 1.0.0 
===

Vastly improves and removes a lot of complexities that caused bugs and misuse of reload.

Added
---

- Two new badges to the README (code-style and npm version)
- Verbose mode as option for both Express and command line usage
- A sample app for express

Modified
---

- Re-wrote the README to reflect all of these changes
- Updated dependencies to their latest version’s
- Fixed race condition that caused reload to spam the server when using sockets for automatic reloading.

Removed
---

- All delays (wait, normal, and socket) (Reload is now all automatic using web sockets (no delays at all))
- Client side sockjs web sockets (removed sockjs)  (Now using native web sockets on the client side and ws on server side)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,54 +2,60 @@ reload
...
-**server.js:**
+**`server.js`:**
 ```javascript
 var express = require('express')
-  , http = require('http')
-  , path = require('path')
-  , reload = require('reload')
-  , bodyParser = require('body-parser')
-  , logger = require('morgan')
+    http = require('http')
```
Comment (jprichardson) [inline README.md]: You'll want to fix this to comply with `standard`.

File: lib/reload-client.js [hunk 2]
```diff
@@ -1,52 +1,99 @@
...
-      }, socketDelay)
+  // The first change flag is used to tell reload to wait until the socket closes at least once before we allow the page to open on a socket open event. Otherwise reload will go into a inifite loop, as the page will have a socket on open event once it loads for the first time
+  var firstChangeFlag = false
+
+  // The navigatedAwayFromPageFlag is set to true in the event handler onbeforeunload because we want to short-circuit reload to prevent it from causing the page to reload before the navigation occurs.
+  var navigatedAwayFromPageFlag = false
+
+    // Wait until the page loads for the first time and then call the webSocketWaiter function so that we can connect the socket for the first time
+  window.addEventListener('load', function () {
+    if (verboseLogging === true) {
```
Comment (jprichardson) [inline lib/reload-client.js]: Can change this to `if (verboseLogging)`.

File: lib/reload-server.js [hunk 3]
```diff
@@ -12,45 +13,59 @@ var dir = process.argv[3]
 var openBrowser = (process.argv[4] === 'true')
 var hostname = process.argv[5]
 var runFile = process.argv[6]
-var reloadDelay = process.argv[7]
-var wait = (process.argv[8] === 'true')
-var startPage = process.argv[9]
+var startPage = process.argv[7]
+var verbose = (process.argv[8] === 'true')
```
Comment (jprichardson) [inline lib/reload-server.js]: I'd suggest taking a look at [`minimist`](https://www.npmjs.com/package/minimist).

**Comment by alallier** [comment]
Updated based on your requests

**Pull request merged**